pkg/appgw/frontend_listeners.go

```diff
@@ -1,16 +1,93 @@
 package appgw
 
-import "k8s.io/api/extensions/v1beta1"
+import (
+	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"k8s.io/api/extensions/v1beta1"
+)
 
-func (builder *appGwConfigBuilder) getListenerConfigs(ingressList []*v1beta1.Ingress) map[frontendListenerIdentifier]*frontendListenerAzureConfig {
-	httpListenersAzureConfigMap := make(map[frontendListenerIdentifier]*frontendListenerAzureConfig)
+// getFrontendListeners constructs the unique set of App Gateway HTTP listeners across all ingresses.
+func (builder *appGwConfigBuilder) getFrontendListeners(ingressList []*v1beta1.Ingress) (*[]n.ApplicationGatewayHTTPListener, map[frontendListenerIdentifier]*n.ApplicationGatewayHTTPListener) {
+	// TODO(draychev): this is for compatibility w/ RequestRoutingRules and should be removed ASAP
+	legacyMap := make(map[frontendListenerIdentifier]*n.ApplicationGatewayHTTPListener)
+
+	var httpListeners []n.ApplicationGatewayHTTPListener
+
+	for listener, config := range builder.getListenerConfigs(ingressList) {
+		var secretFullName string
+		var protocol n.ApplicationGatewayProtocol
+
+		if config != nil {
```

Comment:
Instead of doing a null check, should we populate this in case of Default Listener on #L68

Reply:
!!!